ord_schema/interface/ord_client.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 Open Reaction Database Project Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python API for the Open Reaction Database."""
+
+import urllib.parse
+
+import requests
+
+from ord_schema import message_helpers
+from ord_schema import validations
+from ord_schema.proto import dataset_pb2
+
+TARGET = 'https://client.open-reaction-database.org'
+ORD_DATA_URL = 'https://github.com/Open-Reaction-Database/ord-data/raw/main/'
+
+
+def fetch_dataset(dataset_id):
+    """Fetches a single Dataset message.
+
+    Datasets are first class objects in GitHub, so this method skips the
+    client API and goes directly to the ord-data repository.
+
+    Args:
+        dataset_id: String dataset ID.
+
+    Returns:
+        Dataset message.
+
+    Raises:
+        RuntimeError: The dataset request failed.
+        ValueError: The dataset ID is invalid.
+    """
+    if not validations.is_valid_dataset_id(dataset_id):
+        raise ValueError(f'Invalid dataset ID: {dataset_id}')
+    url = urllib.parse.urljoin(ORD_DATA_URL,
+                               f'{message_helpers.id_filename(dataset_id)}.pb')
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise RuntimeError(
+            f'Request {url} failed with status {response.status_code}')
+    return dataset_pb2.Dataset.FromString(response.content)
+
+
+class OrdClient:
+    """Client for the Open Reaction Database."""
+
+    def __init__(self, target=None):
+        """Initializes the client.
+
+        Args:
+            target: Endpoint URL for client queries. Defaults to the public
+                ORD client URL (defined in TARGET).
+        """
+        if not target:
+            target = TARGET
+        self._target = target
+
+    def fetch_datasets(self, dataset_ids):
+        """Fetches one or more Dataset messages.
+
+        Args:
+            dataset_ids: List of dataset IDs.
+
+        Returns:
+            List of Dataset messages.
+        """
+        # NOTE(kearnes): Return datasets in the same order as requested.
+        datasets = {
+            dataset_id: self.fetch_dataset(dataset_id)
+            for dataset_id in set(dataset_ids)
```

Comment:
Should this be an ordered set to keep things in order?

Reply:
Are you thinking of a specific data structure? I'm not aware of an OrderedSet class, and set()/np.unique() return things in sorted order.

Reply:
Was the intention to sort the results? Based on the comment, I thought your intention was to take return things in the order they were listed in the query.

It doesn't seem like there is a good built-in library with an OrderedSet. There's an `ordered-set` on PyPI but let's not use that. I suppose a hacky solution would be to iterate over `{k:None for k in dataset_ids}.keys()` since dictionaries are ordered. 

I don't think we _have_ to return things in the same query order, but you should revise your comment to say that datasets are returned in a sorted order.

Reply:
Just realized the return statement re-sorts things, never mind.